pyqtgraph/flowchart/Flowchart.py

```diff
@@ -535,7 +536,9 @@ def saveFile(self, fileName=None, startDir=None, suggestedFileName='flowchart.fc
             self.fileDialog.fileSelected.connect(self.saveFile)
             return
             #fileName = QtGui.QFileDialog.getSaveFileName(None, "Save Flowchart..", startDir, "Flowchart (*.fc)")
-        fileName = unicode(fileName)
+        if not fileName.endswith('.fc'):
```

Comment:
If I were to manually call `flowchart.saveFile('filename')`, I would not expect any `.fc` to be added automatically. Can you modify this such that the extension is only added to user-entered file names? You can connect `fileDialog.fileSelected` to a different method (maybe `saveFileSelected` or similar) that does the extension check and then calls `saveFile`.